src/main/java/com/hms_networks/americas/sc/extensions/logging/Logger.java

```diff
@@ -80,6 +82,17 @@ public class Logger {
   /** Queue of unprinted logs. */
   private static LogQueue logQueue;
 
+  /**
+   * List for tracking whether a specific log has been outputted. This is used to prevent spamming
+   * the log with the same message.
+   *
+   * <p>If the log has already been outputted, its key will be in the list, otherwise it will not be
+   * present.
+   *
+   * @since 1.15.9
+   */
+  private static final List logOnceList = new ArrayList(); // List<String>
```

Comment:
I prefer hash map to array for faster lookups.

Reply:
Migrated to a HashSet-based implementation, which in Java is backed by a HashMap.